### PR TITLE
Add option to specify transfer log percentage

### LINKED ISSUE #334: Feature: make transfer_summarizer configurable
See discussion in #133

### PR DESCRIPTION
**Pull request by aubergene**
Fixes #334

### CODE REVIEW

**Comment by aubergene** [comment]
The tests `test_upload`, `test_download` are mocked and don't actually test the real output anyway

**Comment by mattbrictson** [comment]
Hi, thanks for the PR! I think we can go without a test in this case, but I still would like to understand how this feature is intended to be used. The original issue #133 is that `upload!` and `download!` are excessively verbose when dealing with large files. In place of an automated test, could you do a large file upload/download with this branch and show us the actual output that is produced? Thanks!

Also, there is a Travis failure due a RuboCop violation, but that is not related to this PR; you can disregard it.

However as @capistrano-bot [noted above](https://github.com/capistrano/sshkit/pull/401#issuecomment-308400833), I will need a CHANGELOG entry to merge this.

**Comment by mattbrictson** [comment]
If you could also rebase on the latest `master`, that will resolve the RuboCop failure.

**Comment by aubergene** [comment]
Ah, yes actually the fix for #133 already makes things a lot better (I only updated yesterday). The use case for us is that we are uploading to a minimum of 4 servers and as it's both fast and reliable having the Jenkins log filled with lots of uploading lines isn't very helpful, but actually ~40 lines isn't too bad.

Here's an example of it set to 25%

```ruby
upload! 'some_local_file.txt', '/home/some_user/somewhere'
  download! '/home/some_user/some_remote_file.txt', 'somewhere_local', :log_percent 25
```
screen shot 2017-06-15 at 11 34 14 am

I rebased to master, unsure why the Travis job failed for ruby 2.3.3

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ appear at the top.
 
 ## [Unreleased][]
 
-  * Your contribution here!
```
Comment (mattbrictson) [inline CHANGELOG.md]: Please don't remove this line! Thanks

File: lib/sshkit/backends/netssh.rb [hunk 2]
```diff
@@ -96,14 +96,16 @@ def config
...
-      def transfer_summarizer(action)
+      def transfer_summarizer(action, options = {})
+        log_percent = options[:log_percent] || 10
+        log_percent = 100 if log_percent <= 0
         last_name = nil
         last_percentage = nil
         proc do |_ch, name, transferred, total|
           percentage = (transferred.to_f * 100 / total.to_f)
-          unless percentage.nan?
+          unless percentage.nan? || !log_percent
```
Comment (mattbrictson) [inline lib/sshkit/backends/netssh.rb]: Why would `log_percent` be falsey?

**Comment by mattbrictson** [comment]
Thanks for the explanation and screen shot! I think this PR is good to merge after the code comments above are addressed.

**Comment by aubergene** [comment]
Thanks, originally I was going to allow it to be `false` or a number, but it's fine just set it to 100 and you get a single event

**Comment by mattbrictson** [comment]
Thanks! 🙇

**Pull request merged**